### PR TITLE
deprecation warning for old parametric method syntax. fixes #11310

### LINKED ISSUE #11310: fully switch static parameter syntax to `where`
``` jl
julia> type Foo{T}
           bar
       end

julia> Foo{T}() = Foo{T}(1)
Warning: static parameter T does not occur in signature for call at none:1.
The method will not be callable.
Foo{T}

julia> call{T}(::Type{Foo{T}}) = Foo{T}(1)
call (generic function with 941 methods)

julia> Foo{:bar}()
Foo{:bar}(1)
```

i.e. the method definition form fails but the `call` form works. It might be nice if the former was lowered to the latter – having them mean different things is counter-intuitive to me.

Edit: Looks like the issue is the ambiguity of `Foo{T}() = ...` with `call{T}(::Type{Foo{T}}) = ...` and `call{T, S}(::Type{Foo{S}}) = ...`. If there's no way around this perhaps it should raise an ambiguity error with the `call` methods as suggestions.

### PR DESCRIPTION
**Pull request by JeffBezanson**
OK folks, it's time to rip the band-aid off.

### CODE REVIEW

File: src/julia-syntax.scm [hunk 1]
```diff
@@ -810,14 +810,16 @@
                   (sig    (car temp))
                   (params (cdr temp)))
              (if (pair? params)
-                 (let* ((lnos (filter (lambda (e) (and (pair? e) (eq? (car e) 'line)))
-                                      body))
-                        (lno (if (null? lnos) '() (car lnos))))
-                   (syntax-deprecation #f
-                                       (string "inner constructor " name "(...)" (linenode-string lno))
-                                       (deparse `(where (call (curly ,name ,@params) ...) ,@params)))))
+                 (syntax-deprecation #f
```
Comment (tkelman) [inline src/julia-syntax.scm]: either here on in deprecated.jl, should these be annotated with "remove in version ..." ?

**Comment by JeffBezanson** [comment]
Anybody understand the linux64 failure here?

**Comment by yuyichao** [comment]
Seen it once on https://github.com/JuliaLang/julia/pull/22684 last night. No sure what happened and doesn't seem to be clear error message...

**Comment by ararslan** [comment]
Rebased and fixed conflict. We'll see whether the failure shows up again.

**Comment by JeffBezanson** [comment]
Script to automatically fix these:

https://discourse.julialang.org/t/psa-parametric-method-syntax-deprecation-about-to-drop/4931/15
https://gist.github.com/ZacLN/2f0d66c651647e32eae85e58102315cb

Just tried it myself, seems to work well.

**Comment by ararslan** [comment]
Nice! Is this ready to be merged then? AV failure is a timeout.

**Comment by martinholters** [comment]
Resolved conflicts using web interface, so needs squashing.

**Review by Sacha0 — APPROVED** [review]
:) !

**Comment by quinnj** [comment]
Oh goody, let's see if I missed any from my pre-emptive strike.

**Pull request merged**